runtime/domain.c

```diff
@@ -89,6 +97,9 @@ struct dom_internal {
   struct domain state;
   struct interruptor interruptor;
 
+  /* remark stack */
+  struct remark_stack remark_stack;
```

Comment:
So the `remark_stack` for a domain has to be globally accessible independent of the domain lifecycle. Right now the another structure like that is the `interruptor` which is why the `remark_stack` ended up there. I was very hesitant to have things in `Caml_state` which are independent of domain lifecycle and other domains might read/write. 

I feel the `remark_stack` should belong in a global structure in `major_gc.c` where it is used to avoid it being spread out into `domain.c`.